packages/@react-stately/tooltip/test/useTooltipTriggerState.test.js

```diff
@@ -0,0 +1,200 @@
+ * governing permissions and limitations under the License.
+ */
+
+import {act, fireEvent, render} from '@react-spectrum/test-utils';
+import {mergeProps, useTooltip, useTooltipTrigger} from 'react-aria';
+import React from 'react';
+import {useTooltipTriggerState} from '../src';
+
+// Sync with TOOLTIP_COOLDOWN & TOOLTIP_DELAY in useTooltipTriggerState
+const TOOLTIP_COOLDOWN = 500;
+const TOOLTIP_DELAY = 1500;
+
+function Tooltip({state, ...props}) {
+  let {tooltipProps} = useTooltip(props, state);
+
+  return (
+    <span {...mergeProps(props, tooltipProps)}>
+      {props.children}
+    </span>
+  );
+}
+
+function TooltipTrigger(props) {
+  let state = useTooltipTriggerState(props);
+  let ref = React.useRef();
+
+  let {triggerProps, tooltipProps} = useTooltipTrigger(props, state, ref);
+
+  return (
+    <span>
+      <button aria-label="trigger" ref={ref} {...triggerProps}>
+        {props.children}
+      </button>
+      {state.isOpen &&
+        <Tooltip state={state} {...tooltipProps}>{props.tooltip}</Tooltip>}
+    </span>
+  );
+}
+
+/**
+ * Most of the tests for useTooltipTriggerState are in the React Spectrum package at
+ * @react-spectrum/tooltip/test/TooltipTrigger.test.js. The React Spectrum tooltip
+ * does not implement a custom close delay.
+ */
+describe('useTooltipTriggerState', function () {
+  let onOpenChange = jest.fn();
+
+  beforeAll(() => {
+    jest.useFakeTimers('legacy');
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  afterEach(() => {
+    onOpenChange.mockClear();
+    // there's global state, so we need to make sure to run out the cooldown for every test
+    act(() => {jest.runAllTimers();});
+  });
+
+  describe('custom close delay', () => {
+    it('closes with a custom close delay less than the default', () => {
+      let closeDelay = 100;
+  
+      let {getByRole, queryByRole, getByLabelText} = render(
+        <TooltipTrigger onOpenChange={onOpenChange} closeDelay={closeDelay} tooltip="Helpful information">
+          Trigger
+        </TooltipTrigger>
+      );
+      fireEvent.mouseDown(document.body);
+      fireEvent.mouseUp(document.body);
+
+      let button = getByLabelText('trigger');
+      fireEvent.mouseEnter(button);
+      fireEvent.mouseMove(button);
+
+      expect(onOpenChange).not.toHaveBeenCalled();
+      expect(queryByRole('tooltip')).toBeNull();
+
+      // run through open timer and confirm that it has opened
+      act(() => jest.advanceTimersByTime(TOOLTIP_DELAY));
+      expect(onOpenChange).toHaveBeenCalledWith(true);
+      let tooltip = getByRole('tooltip');
+      expect(tooltip).toBeVisible();
+
+      fireEvent.mouseLeave(button);
+      fireEvent.mouseMove(button);
+
+      // run the rest of the way through the timer and confirm that it has closed
+      // Note that closeDelay is less than the default close time of 500ms
+      act(() => jest.advanceTimersByTime(closeDelay / 2));
+      expect(onOpenChange).toHaveBeenCalledWith(true);
+    });
```

Comment:
We haven't tested that advancing the rest of the time (aka the other half of `closeDelay`) would cause `onOpenChange` to fire with `false` and that the tooltip would disappear in this test.

Reply:
👍